libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftClients.java

```diff
@@ -88,9 +96,19 @@ private static TProtocol makeProtocol(String url, String service) {
         THttpClient thriftClient = null;
         final String destinationAddress = url + service;
         try {
-            thriftClient = new THttpClient(destinationAddress);
+            if (BACKEND_PROXY_URL != null) {
+                URL proxyUrl = new URL(BACKEND_PROXY_URL);
+                HttpHost proxy = new HttpHost(proxyUrl.getHost(), proxyUrl.getPort(), proxyUrl.getProtocol());
+                DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(proxy);
+                CloseableHttpClient httpClient = HttpClients.custom().setRoutePlanner(routePlanner).build();
+                thriftClient = new THttpClient(destinationAddress, httpClient);
+            } else {
+                thriftClient = new THttpClient(destinationAddress);
+            }
         } catch (TTransportException e) {
             log.error("cannot connect to backend on " + destinationAddress, e);
+        } catch (MalformedURLException e) {
+            log.error("cannot connect via http proxy (REASON:MalformedURLException) to thrift backend:  " + BACKEND_PROXY_URL, e);
```

Comment:
does the `BACKEND_PROXY_URL` maybe contain basic auth credentials (in some configurations)? In that case it might better not to log it.